api/fastapi_service/main.py

```
@@ -96,60 +101,85 @@ async def city_regions(
+        return graphBase
+
+    except HTTPException:
+        # Re-raise HTTP exceptions without modification
+        raise
+    except Exception as e:
+        # Catch any other unexpected errors and provide detailed error message
+        status_code = 500
+        error_type = type(e).__name__
+        error_msg = str(e)
+        detail = f"Internal server error processing graph for city {city_id}, regions {regions_ids}. Error: {error_type}: {error_msg}"
         logger.error(f"{request} {status_code} {detail}")
```

Ruff (0.14.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)